src/main/java/org/zalando/fahrschein/NakadiClient.java

```diff
@@ -80,33 +86,30 @@ public Subscription subscribe(String applicationName, String eventName, String c
     }
 
     public <T> void listen(Subscription subscription, Class<T> eventType, Listener<T> listener, StreamParameters streamParameters) throws IOException {
-        listen(subscription, eventType, listener, streamParameters, NO_METRICS_COLLECTOR);
-    }
-
-    public <T> void listen(Subscription subscription, Class<T> eventType, Listener<T> listener, StreamParameters streamParameters, @Nullable MetricsCollector metricsCollector) throws IOException {
         final String eventName = Iterables.getOnlyElement(subscription.getEventTypes());
         final String queryString = streamParameters.toQueryString();
         final URI uri = baseUri.resolve(String.format("/subscriptions/%s/events?%s", subscription.getId(), queryString));
 
-        final NakadiReader<T> nakadiReader = nakadiReaderFactory.createReader(uri, eventName, Optional.of(subscription), eventType, listener, metricsCollector);
+        final NakadiReader<T> nakadiReader = nakadiReaderFactory.createReader(uri, eventName, Optional.of(subscription), eventType, listener);
 
-        nakadiReader.run(streamParameters.getStreamTimeout().orElse(0), TimeUnit.SECONDS);
-    }
-
-    public <T> void listen(String eventName, Class<T> eventType, Listener<T> listener) throws IOException {
-        listen(eventName, eventType, listener, new StreamParameters());
+        nakadiReader.run();
     }
 
     public <T> void listen(String eventName, Class<T> eventType, Listener<T> listener, StreamParameters streamParameters) throws IOException {
-        listen(eventName, eventType, listener, streamParameters, NO_METRICS_COLLECTOR);
+        listen(eventName, eventType, listener, streamParameters, -1, TimeUnit.SECONDS);
     }
 
-    public <T> void listen(String eventName, Class<T> eventType, Listener<T> listener, StreamParameters streamParameters, @Nullable MetricsCollector metricsCollector) throws IOException {
+    public <T> void listen(String eventName, Class<T> eventType, Listener<T> listener, StreamParameters streamParameters, long lockTimeout, TimeUnit timeoutUnit) throws IOException {
         final String queryString = streamParameters.toQueryString();
         final URI uri = baseUri.resolve(String.format("/event-types/%s/events?%s", eventName, queryString));
 
-        final NakadiReader<T> nakadiReader = nakadiReaderFactory.createReader(uri, eventName, Optional.<Subscription>empty(), eventType, listener, metricsCollector);
+        final NakadiReader<T> nakadiReader = nakadiReaderFactory.createReader(uri, eventName, Optional.<Subscription>empty(), eventType, listener);
+
+        if (lockTimeout > 0) {
+            final Thread currentThread = Thread.currentThread();
+            executor.schedule(currentThread::interrupt, lockTimeout, timeoutUnit);
```

Comment:
> Out of curiosity, why don't you use stream_timeout?

I want to have control of the timeout on client side. Using the same value for stream and lock timeout makes sense though.